algorithms/C/searching/Interpolation-search.c

```diff
@@ -0,0 +1,69 @@
+/*
+Program to search data in array using interpolation search
+Interpolation search: Used to search data in a sequence stored array
+*/
+
+#include<stdio.h>
+#include<stdlib.h>
+// Function to search data
+int interpolation_search(int *arr,int low,int high,int to_search)
+{
+    if (low<=high&&to_search>=arr[low]&&to_search<= arr[high])
+    {
+    // formula to find index
+int pos=low+(to_search-arr[low])*(high-low)/(arr[high]-arr[low]);
+
+    // if data found
+    if(arr[pos]==to_search)
+        return pos;
+    // if data is less than current position
+    else if(arr[pos]>to_search)
+        high=pos-1;
+    // if data is greater than the current position
+    else if(arr[pos]<to_search)
+        low=pos+1;
+    // recursive call
+    return interpolation_search(arr,low,high,to_search);
+    }
+    // return -1 if the if condition fails
+    return -1;
+}
+//driver code
+int main()
+{
+    int *arr,size,index,to_search,i;
+
+    printf("Enter the size: ");
+    // taking size
+    scanf("%d",&size);
+    // dynamically locating memory
+    arr=(int*)malloc(sizeof(int)*size);
+    for(i=0;i<size;i++)
+    {
+        scanf("%d",&arr[i]);
```

Comment:
**Possible null dereference [CWE-690]:** 

`arr` will point to `NULL` in case malloc() fails to allocate the memory ( maybe due to insufficient memory or memory corruption ) 

**Fix:**
perform a check before entering the loop for allocation failure.